sixpack.js

```diff
@@ -215,17 +232,20 @@
                         }
                     }
 
-                    if (!timed_out) {
-                        clearTimeout(timeout_handle);
-                        return callback(null, data);
-                    }
+                    _log_request_time_end(debug, 'SixpackSession - participate request duration');
+                    _log(debug, 'SixpackSession - Request Ended: ', data);
+                    return callback(null, data);
                 });
             });
+            req.on('timeout', function () {
+                _log(debug, `SixpackSession - Request Timeout - ${options.timeout}ms expired.`);
+                req.destroy();
```

Comment:
Instead of call `.destroy()`, you could `return callback(new Error('request timed out'));` as you already know the timeout occurred